kratos/python_scripts/sympy_fe_utilities.py

```diff
@@ -1,466 +1,522 @@
+        for i in range(nnodes):
+            for _ in range(dim):
+                B[0, i*dim] = DN[i,0]
+                B[0, i*dim+1] = 0
+                B[1, i*dim] = 0
+                B[1, i*dim+1] = DN[i,1]
+                B[2, i*dim] = DN[i,1]
+                B[2, i*dim+1] = DN[i,0]
+    elif dim == 3:
         strain_size = 6
         nnodes = DN.shape[0]
-        B = Matrix( zeros(strain_size, nnodes*dim) )
-        for i in range(0,nnodes):
-            B[ 0, i*3 ] = DN[ i, 0 ];
-            B[ 1, i*3 + 1 ] = DN[ i, 1 ];
-            B[ 2, i*3 + 2 ] = DN[ i, 2 ];
-            B[ 3, i*3 ] = DN[ i, 1 ];
-            B[ 3, i*3 + 1 ] = DN[ i, 0 ];
-            B[ 4, i*3 + 1 ] = DN[ i, 2 ];
-            B[ 4, i*3 + 2 ] = DN[ i, 1 ];
-            B[ 5, i*3 ] = DN[ i, 2 ];
-            B[ 5, i*3 + 2 ] = DN[ i, 0 ];
+        B = sympy.Matrix(sympy.zeros(strain_size, nnodes*dim))
+        for i in range(nnodes):
+            B[0, i*3 ] = DN[i, 0]
+            B[1, i*3 + 1] = DN[i, 1]
+            B[2, i*3 + 2] = DN[i, 2]
+            B[3, i*3] = DN[i, 1]
+            B[3, i*3 + 1] = DN[i, 0]
+            B[4, i*3 + 1] = DN[i, 2]
+            B[4, i*3 + 2] = DN[i, 1]
+            B[5, i*3] = DN[i, 2]
+            B[5, i*3 + 2] = DN[i, 0]
     else:
-        print("dimension asked in Matrix B is ",dim)
-        raise Exception("wrong dimension")
+        print("dimension asked in Matrix B is ", dim)
+        raise ValueError("wrong dimension")
     return B
 
 def grad_sym_voigtform(DN, x):
-    """ This method defines a symmetric gradient
+    """
+    This method defines a symmetric gradient.
 
     Keyword arguments:
-    DN -- The shape function derivatives
-    x -- The variable to compute the gradient
+    - DN -- The shape function derivatives
+    - x -- The variable to compute the gradient
     """
     [nnodes, dim] = x.shape
 
     B = MatrixB(DN)
 
     # Put the x components one after the other in a vector
-    xvec = Matrix( zeros(B.shape[1], 1 ) );
-    for i in range(0,nnodes):
-        for k in range(0,dim):
+    xvec = sympy.Matrix(sympy.zeros(B.shape[1], 1))
+    for i in range(nnodes):
+        for k in range(dim):
             xvec[i*dim+k] = x[i,k]
 
-    return simplify( B*xvec )
-
-def grad(DN,x):
-    """ This method defines a gradient
-
-    Keyword arguments:
-    DN -- The shape function derivatives
-    x -- The variable to compute the gradient
-    """
-    error_msg = """
-                    The function grad(DN,x) was removed to avoid misunderstandings.
-                    You should use either DfjDxi(DN,f) or DfiDxj(DN,f).
-
-                    DfjDxi(DN,f): returns a matrix D such that D(i,j) = D(fj)/D(xi) - the standard one in fluid dynamics
-                    that is:
-                        D(f1)/D(x1) D(f2)/D(x1) D(f3)/D(x1)
-                        D(f1)/D(x2) D(f2)/D(x2) D(f3)/D(x2)
-                        D(f1)/D(x3) D(f2)/D(x3) D(f3)/D(x3)
```

Comment:
I'd add this documentation to `DfjDxi` method. I remember struggling a bit with this in the past.